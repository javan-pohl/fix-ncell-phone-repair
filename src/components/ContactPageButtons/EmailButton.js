import * as React from 'react'
import { contactButton } from './ContactPage.module.css'
import emailIcon from '../../images/email.png'

const EmailButton = () => (
  <button type="button" className={contactButton}>
    <a data-testid="emailButtonA" href="mailto:fixncellllc@gmail.com">
      <img src={emailIcon} alt="email icon" />
      <br />
      Email Us
    </a>
  </button>
)

export default EmailButton
