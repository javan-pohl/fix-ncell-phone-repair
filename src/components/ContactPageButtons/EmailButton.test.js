import 'jsdom-global/register'
import '@testing-library/jest-dom'
import React from 'react'
import * as Gatsby from 'gatsby'
import renderer from 'react-test-renderer'
import {render, fireEvent} from '@testing-library/react'
import "@testing-library/jest-dom/extend-expect"

import EmailButton from './EmailButton'
import emailIcon from '../../images/email.png'

const useStaticQuery = jest.spyOn(Gatsby, 'useStaticQuery')
useStaticQuery.mockImplementation(() => ({
  site: {
    siteMetadata: {
      email: 'fixncellllc@gmail.com',
    },
  },
}))

beforeEach(() => {
  jest.clearAllMocks()
})

describe('EmailButton', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<EmailButton />).toJSON()
    expect(tree).toMatchSnapshot()
  })
	it('should have href attribute "mailto:fixncellllc@gmail.com"', () => {
		const {queryByTestId} = render(<EmailButton />)
		const emailButton = queryByTestId("emailButtonA")
		expect(emailButton).toHaveAttribute('href', 'mailto:fixncellllc@gmail.com')
	})
})


