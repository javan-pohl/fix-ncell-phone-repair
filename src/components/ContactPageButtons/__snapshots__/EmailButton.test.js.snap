// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmailButton renders correctly 1`] = `
<button
  className="contactButton"
  type="button"
>
  <a
    data-testid="emailButtonA"
    href="mailto:fixncellllc@gmail.com"
  >
    <img
      alt="email icon"
      src=""
    />
    <br />
    Email Us
  </a>
</button>
`;
